Allow the Current Ruleset to be changed

frank-weinberg opened the pull request:
This allows e.g. changing intermission duration or en/disabling auto start/stop rules without a reset of the teams.

frank-weinberg committed:
Allow the Current Ruleset to be changed

brian-brazil commented on src/com/carolinarollergirls/scoreboard/core/impl/ClockImpl.java:
```diff
@@ -62,6 +63,15 @@ else if (flag != Flag.INTERNAL) {
             return getMaximumTime() - getTime();
         }
+        if (prop == Value.MAXIMUM_TIME && getId() == ID_INTERMISSION) {
+            long duration = DEFAULT_MAXIMUM_TIME;
```
This means I can no longer override the maximum time for an intermissions from the operator screen, which is often required when the schedule needs moving forward.

brian-brazil commented:
That updating/deleting rulesets that are in use will affect the in-progress game may not be completely obvious to users.

frank-weinberg committed:
review comments

frank-weinberg commented:
I have removed the delete button (because it's completely unclear what should be done if the active ruleset is deleted) and added a note that editing will affect the current game for rulesets currently in effect.
(I've also changed handling of max intermission time so manual overrides are possible again.)

brian-brazil commented:
That works.

A few small issues:
- Changing the current ruleset dosen't update the Delete button/warning
- If you create a ruleset that's only whitespace, things get odd
- The Name of "WFTDA Sanctioned" shouldn't be in a textbox, as it's not editable

I suspect the last two are existing.

frank-weinberg committed:
more fixes

frank-weinberg commented:
They were existing but sufficiently easy fixes/workarounds.

brian-brazil commented on html/components/rulesets-tab.js:
```diff
@@ -261,6 +302,9 @@ function createRulesetsTab(tab) {
 				WS.Set("ScoreBoard.Rulesets.Ruleset(" + activeRuleset.Id + ").Rule(" + val.Fullname + ")", value);
 			});
+			var newName = tab.find("#name").val();
+			if (newName.trim() === '') { newName = 'Unnamed'; }
```
Elsewhere (colors) we ignore such requests, I think we should do the same here.

brian-brazil commented on html/components/rulesets-tab.js:
```diff
@@ -225,10 +258,17 @@ function createRulesetsTab(tab) {
 		});
 
 		if (rs.Immutable) {
+			tab.find("#name").attr('readonly', true);
```
I think disabled would be a bit clearer

brian-brazil reviewed (commented):
Grand, saves me from fixing them.

frank-weinberg committed:
review comments & bugfix

frank-weinberg commented:
Addressed the last comments and fixed an NPE discovered in the process.
